string[] robot_list
string[] task_list
